### PR TITLE
Feature/tclune/mapl3 wildcard support

### PR DESCRIPTION
**Pull request by tclune**
## Description

Added support for using wildcard in history via generic3g features.

Tested with unit tests.  Not with model.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [ ] I have tested this change with a run of GEOSgcm (if non-trivial)
- [ ] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [ ] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

File: generic3g/connection/VirtualConnectionPt.F90 [hunk 1]
```diff
@@ -199,4 +214,31 @@ subroutine write_formatted(this, unit, iotype, v_list, iostat, iomsg)
...
+!!$      _HERE, item%get_full_name()
+!!$      _HERE, matches
+
+!!$      if (this%get_full_name() == '*') then
+!!$         matches = .true.
+!!$         return
+!!$      end if
+!!$      matches = () .and. &
+!!$                (this%get_full_name() == item%get_full_name())
+
```
[the remark this answers is not in the document]
Comment (atrayano) [reply generic3g/connection/VirtualConnectionPt.F90:241]: My bad, I tried to suggest to remove these lines (they are commented out, but should they be there), but messed up

**Review by atrayano — DISMISSED** [review]

File: generic3g/connection/VirtualConnectionPt.F90 (hunk 1 above)
[the remark this answers is not in the document]
Comment (tclune) [reply generic3g/connection/VirtualConnectionPt.F90:241]: No you were correct and should have suggested ala:
```suggestion
```

**Review by atrayano — DISMISSED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -176,21 +176,21 @@ endif ()
...
+#   string(REGEX REPLACE "v8.5.0b([0-9]+)" "\\1" ESMF_BETA_SNAPSHOT_NUMBER ${ESMF_BETA_SNAPSHOT})
+#   if (ESMF_BETA_SNAPSHOT_NUMBER LESS ESMF_BETA_SNAPSHOT_TARGET)
+#     message(FATAL_ERROR
+#       "ERROR! ESMF version must be at least v8.5.0b22, but you are using ${ESMF_BETA_SNAPSHOT}\n"
+#       ""
+#       "This is due to the use of a feature of ESMF that came in with ESMF v8.5.0b22, a beta version of ESMF.\n"
+#       "This is a temporary fix until stable ESMF 8.5.0 is released.\n"
+#       )
+#   endif ()
+# endif ()
```
Comment (tclune) [inline CMakeLists.txt:193]:
```suggestion
 if (ESMF_VERSION VERSION_EQUAL 8.5.0 AND ESMF_BETA_RELEASE)
   # So now we are using a beta version of ESMF 8.5.0. We need to make sure
   # that the version is at least 8.5.0b22. That version information
   # is stored in ESMF_BETA_SNAPSHOT and is of the form "v8.5.0b22"
   set (ESMF_BETA_SNAPSHOT_TARGET 22)
   string(REGEX REPLACE "v8.5.0b([0-9]+)" "\\1" ESMF_BETA_SNAPSHOT_NUMBER ${ESMF_BETA_SNAPSHOT})
   if (ESMF_BETA_SNAPSHOT_NUMBER LESS ESMF_BETA_SNAPSHOT_TARGET)
     message(FATAL_ERROR
       "ERROR! ESMF version must be at least v8.5.0b22, but you are using ${ESMF_BETA_SNAPSHOT}\n"
       ""
       "This is due to the use of a feature of ESMF that came in with ESMF v8.5.0b22, a beta version of ESMF.\n"
       "This is a temporary fix until stable ESMF 8.5.0 is released.\n"
       )
   endif ()
 endif ()
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -176,21 +176,21 @@ endif ()
...
+   string(REGEX REPLACE "v8.5.0b([0-9]+)" "\\1" ESMF_BETA_SNAPSHOT_NUMBER ${ESMF_BETA_SNAPSHOT})
+   if (ESMF_BETA_SNAPSHOT_NUMBER LESS ESMF_BETA_SNAPSHOT_TARGET)
+     message(FATAL_ERROR
+       "ERROR! ESMF version must be at least v8.5.0b22, but you are using ${ESMF_BETA_SNAPSHOT}\n"
+       ""
+       "This is due to the use of a feature of ESMF that came in with ESMF v8.5.0b22, a beta version of ESMF.\n"
+       "This is a temporary fix until stable ESMF 8.5.0 is released.\n"
+       )
+   endif ()
+ endif ()
```
[the remark this answers is not in the document]
Comment (mathomp4) [reply CMakeLists.txt]: Oops. Did this by hand!

**Pull request merged**